WordPress/src/main/java/org/wordpress/android/util/encryptedlogging/EncryptedLogWriter.kt

```diff
@@ -0,0 +1,95 @@
+package org.wordpress.android.util.encryptedlogging
+
+import android.util.Base64
+import com.goterl.lazycode.lazysodium.interfaces.SecretStream
+import com.goterl.lazycode.lazysodium.utils.Key
+import java.io.File
+import java.io.FileWriter
+import java.util.UUID
+
+/**
+ * EncryptedLogWriter creates encrypted logs.
+ **
+ * @param file A file object representing the log file destination.
+ * @param publicKey The public key used to encrypt the log.
+ * @param uuid (Optional) The UUID string associated with the encrypted log. If one is not provided, it is generated for you.
+ * @constructor Creates the `EncryptedLogWriter`. Call `write` to append data, and `close` when to write out the postamble.
+ *
+ */
+class EncryptedLogWriter(
+    val file: File,
+    publicKey: Key,
+    private val uuid: String = UUID.randomUUID().toString()
+) {
+    private val fileWriter: FileWriter = FileWriter(file)
+    private val sodium = EncryptionUtils.sodium
+    private var state = SecretStream.State.ByReference()
+
+    init {
+        var header = ByteArray(SecretStream.HEADERBYTES)
+        val key = SecretStreamKey.generate()
+        check(sodium.cryptoSecretStreamInitPush(state, header, key.bytes))
+        writeHeader(key.encrypt(publicKey), header)
+    }
+
+    /**
+     * Encrypt and write the provided string to the encrypted log file.
+     * @param String: The string to be written to the file.
+     */
+    fun write(string: String) {
+        val encryptedString = encryptMessage(string, SecretStream.TAG_MESSAGE)
+        fileWriter.write("\t\t\"$encryptedString\",\n")
```

Comment:
This implementation uses the [same streaming write approach as on iOS](https://github.com/Automattic/Automattic-Tracks-iOS/pull/120), because it can be much more memory-efficient without much more complexity. 

The manually-create-the-json stuff is covered by a test, so IMHO the little bit of added complexity is worth it for the efficiency, but it was discussed in that PR, so I figured I'd mention it in this one as well.

Reply:
I am totally good with this approach for this specific usecase 👍